tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Unit\EventSubscriber;
+
+use Apigee\Edge\Exception\ApiException;
+use Drupal\apigee_edge\EventSubscriber\EdgeExceptionSubscriber;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\Core\Config\Config;
+use Prophecy\Argument;
+use Psr\Log\LoggerInterface;
+use stdClass;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Test EdgeExceptionSubscriber.
+ *
+ * @group apigee_edge
+ */
+class EdgeExceptionSubscriberTest extends UnitTestCase {
+
+  protected $exception;
+  protected $http_kernel;
+  protected $logger;
+  protected $redirect_destination;
+  protected $access_unaware_router;
+  protected $config_factory;
+  protected $messenger;
+  protected $get_response_for_exception_event;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->exception = new ApiException("API response error message.");
+
+    $this->http_kernel = $this->prophesize(HttpKernelInterface::class);
+    $this->logger = $this->prophesize(LoggerInterface::class);
+    $this->redirect_destination = $this->prophesize(RedirectDestinationInterface::class);
+
+    $request = $this->prophesize(RequestContext::class);
+    $this->access_unaware_router = $this->prophesize(UrlMatcherInterface::class);
+    $this->access_unaware_router->getContext(Argument::any())->willReturn($request->reveal());
+
+    $this->messenger = $this->prophesize(MessengerInterface::class);
+
+    $request = $this->prophesize(Request::class);
```

Comment:
Maybe mocking with `$this->getMockbuilder(Request::class)->disableOriginalConstructor()->getMock()` would be easier.